### ISSUE TITLE
Can't build the engine with fatal error C1083

### ISSUE THREAD

**Issue by kreaninw** [o3de/o3de#5158]
**Describe the bug**
I have followed the building instruction [here](https://o3de.org/docs/welcome-guide/setup/setup-from-github/). All the prerequisites have been met. However, when I tried to build the engine's binary using cmake:

`cmake --build build/windows_vs2019 --target Editor --config profile -- /m `

I always get this error:

```
Microsoft (R) Build Engine version 16.11.1+3e40a09f8 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  unity_22_cxx.cxx
  unity_19_cxx.cxx
  unity_23_cxx.cxx
  unity_17_cxx.cxx
  unity_25_cxx.cxx
  unity_20_cxx.cxx
  unity_18_cxx.cxx
  unity_21_cxx.cxx
C:/o3de/o3de/Code/Framework/AzCore/AzCore/Utils/TypeHash.cpp(10,10): fatal error C1083: Cannot open include file: 'city.h': No such fi
le or directory [C:\o3de\o3de\build\windows_vs2019\Code\Framework\AzCore\AzCore.vcxproj]
  #include <city.h>
           ^
C:\o3de\o3de\Code\Framework\AzCore\AzCore/JSON/rapidjson.h(54,10): fatal error C1083: Cannot open include file: 'rapidjson/rapidjson.h
': No such file or directory [C:\o3de\o3de\build\windows_vs2019\Code\Framework\AzCore\AzCore.vcxproj]
  #include <rapidjson/rapidjson.h>
           ^
C:\o3de\o3de\Code\Framework\AzCore\AzCore/JSON/rapidjson.h(54,10): fatal error C1083: Cannot open include file: 'rapidjson/rapidjson.h
': No such file or directory [C:\o3de\o3de\build\windows_vs2019\Code\Framework\AzCore\AzCore.vcxproj]
  #include <rapidjson/rapidjson.h>
           ^
C:\o3de\o3de\Code\Framework\AzCore\AzCore/JSON/rapidjson.h(54,10): fatal error C1083: Cannot open include file: 'rapidjson/rapidjson.h
': No such file or directory [C:\o3de\o3de\build\windows_vs2019\Code\Framework\AzCore\AzCore.vcxproj]
  #include <rapidjson/rapidjson.h>
           ^
C:\o3de\o3de\Code\Framework\AzCore\AzCore/JSON/rapidjson.h(54,10): fatal error C1083: Cannot open include file: 'rapidjson/rapidjson.h
': No such file or directory [C:\o3de\o3de\build\windows_vs2019\Code\Framework\AzCore\AzCore.vcxproj]
  #include <rapidjson/rapidjson.h>
           ^
C:\o3de\o3de\Code\Framework\AzCore\AzCore/Script/lua/lua.h(12,13): fatal error C1083: Cannot open include file: 'Lua/lua.h': No such f
ile or directory [C:\o3de\o3de\build\windows_vs2019\Code\Framework\AzCore\AzCore.vcxproj]
  #   include <Lua/lua.h>
              ^
C:\o3de\o3de\Code\Framework\AzCore\AzCore/JSON/rapidjson.h(54,10): fatal error C1083: Cannot open include file: 'rapidjson/rapidjson.h
': No such file or directory [C:\o3de\o3de\build\windows_vs2019\Code\Framework\AzCore\AzCore.vcxproj]
  #include <rapidjson/rapidjson.h>
           ^
C:\o3de\o3de\Code\Framework\AzCore\AzCore/XML/rapidxml.h(17,10): fatal error C1083: Cannot open include file: 'rapidxml/rapidxml.h': N
o such file or directory [C:\o3de\o3de\build\windows_vs2019\Code\Framework\AzCore\AzCore.vcxproj]
  #include <rapidxml/rapidxml.h>
           ^
```

**Failure type**
Build

**To Reproduce Test Failures**
 - `cmake --build build/windows_vs2019 --target Editor --config profile -- /m `

**Expected behavior**
There shouldn't be an error and I should get the binary file to run the engine.

**Screenshots**
![Screenshot 2021-11-01 011312](https://user-images.githubusercontent.com/1403194/139620789-4564aba8-aa57-4f82-bcf3-f4719a156d70.jpg)

**Logs**
[Jenkins.zip](https://github.com/o3de/o3de/files/7450711/Jenkins.zip)

**Desktop/Device (please complete the following information):**
 - Device: PC 
 - OS: Windows
 - Version: 10
 - CPU: Intel i5-8250U
 - GPU: Intel UHD Graphics 620
 - Memory: 8GB

**Additional context**
-

**Comment by johnsdav** [comment]
I also have hit this error following setup steps for both `source` and `engine as sdk` methods. Running this on windows 11 and CMake 3.22.0-rc2

**Comment by Xanadar89** [comment]
I have the same error
This problem occure becouse of the qoutes
If you open project properties ->C/C++-> All Options and look at AdditionalOptions you will see something like this
`%(AdditionalOptions) /Zc:__cplusplus /Zc:lambda /favor:AMD64 /bigobj /RTCsu /experimental:external "-external:I <3rdPartyPackagesFolder>/Lua-5.3.5-rev5-windows/lua/include" "-external:I <3rdPartyPackagesFolder>/RapidJSON-1.1.0-rev1-multiplatform/RapidJSON/include"`

If you remove all qoutes around -external:I and path it would solve the problem
`%(AdditionalOptions) /Zc:__cplusplus /Zc:lambda /favor:AMD64 /bigobj /RTCsu /experimental:external -external:I <3rdPartyPackagesFolder>/Lua-5.3.5-rev5-windows/lua/include -external:I <3rdPartyPackagesFolder>/RapidJSON-1.1.0-rev1-multiplatform/RapidJSON/include ...`

What is left is to figure out why this quotes there in the first place

**Comment by amznestebanpapp** [comment]
Documentation was modified to point to CMake releases. We have seen multiple issues with release candidate versions of CMake. There is a current issue https://github.com/o3de/o3de/issues/5168 to validate 3.22 and fix issues.
Apparently there were changes in how system include paths are handled.

For the time being stay on CMake 3.21.X

**Comment by kreaninw** [comment]
This issue with CMake 3.22rc is fixed in [PR #5314](https://github.com/o3de/o3de/pull/5314). I can compile the engine without any error.
